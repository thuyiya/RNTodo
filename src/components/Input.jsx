import React, { useState, useEffect, useRef } from "react";
import {
  Platform,
  View,
  Text,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  Dimensions,
} from "react-native";
import PropTypes from 'prop-types';
import { theme } from "../constants";
import Typography from "./Typography";

const { width: SCREEN_WIDTH } = Dimensions.get("screen");
const BUTTON_WIDTH = 100;

const Input = ({ onChange, value }) => {
  const [text, setText] = useState("");
  const inputRef = useRef(null);

  const add = () => {
    if (text == "") {
      return;
    }
    onChange(text);
    setText("");
  };

  const keybordApear = () => {
    setText(value);
    if(value !== "") {
      inputRef.current.focus();
    }
  }

  useEffect(() => {
    keybordApear()
  }, [value]);

  return (
    <View style={styles.container}>
      <TextInput
        ref={inputRef}
        placeholder={"Enter Here..."}
        style={styles.input}
        value={text}
        clearButtonMode={"while-editing"}
        onChangeText={(text) => setText(text)}
      />
      <TouchableOpacity style={styles.button} onPress={() => add()}>
        <Typography size="h6" weight="lightBold" color={theme.colors.buttonText} >{value === "" ? "ADD" : "UPDATE"}</Typography>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: theme.colors.white,
    borderRadius: 8,
    margin: 8,
    height: 60,
    width: SCREEN_WIDTH - 16,
  },
  input: {
    width: SCREEN_WIDTH - BUTTON_WIDTH,
    padding: 10,
    height: Platform.OS === "ios" ? 40 : 60,
  },
  button: {
    backgroundColor: theme.colors.primary,
    borderRadius: 18,
    height: 40,
    minWidth: 60,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 8,
  },
});

Input.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func
}

export default Input;
