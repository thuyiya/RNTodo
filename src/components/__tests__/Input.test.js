import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react-native";
import Input from "../Input";

describe("<Input />", () => {
  it("has 2 children", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it("on change text", () => {
    const onChangeTextMock = jest.fn();
    const CHANGE_TEXT = "content";
    render(<Input onChange={onChangeTextMock}/>);
    fireEvent.changeText(screen.getByTestId("input1"), CHANGE_TEXT);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
