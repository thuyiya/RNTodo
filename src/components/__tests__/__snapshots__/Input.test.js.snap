// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "margin": 8,
      "width": 734,
    }
  }
>
  <TextInput
    clearButtonMode="while-editing"
    onChangeText={[Function]}
    placeholder="Enter Here..."
    style={
      {
        "height": 40,
        "padding": 10,
        "width": 650,
      }
    }
    testID="input1"
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#0145a6",
        "borderRadius": 18,
        "height": 40,
        "justifyContent": "center",
        "minWidth": 60,
        "opacity": 1,
        "padding": 8,
      }
    }
    testID="button3"
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 14,
            "fontWeight": "500",
            "margin": 0,
            "textAlign": "center",
          },
          {},
        ]
      }
    >
      UPDATE
    </Text>
  </View>
</View>
`;
